src/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab.tsx

```diff
@@ -1,94 +1,339 @@
-import React, { FC } from "react";
+import React, { FC, useState } from "react";
 
 import Box from "@/components/Box/Box";
 import Flex from "@/components/Flex/Flex";
 import P, { Heading } from "@/components/Typography";
 import Card from "@/components/Card/Card";
-import DropdownSelect from "@/components/DropdownSelect/DropdownSelect";
 import { CurriculumUnitsTabData } from "@/node-lib/curriculum-api-2023";
+import Icon from "@/components/Icon/Icon";
+import OutlineHeading from "@/components/OutlineHeading/OutlineHeading";
+import OakLink from "@/components/OakLink/OakLink";
+import Button from "@/components/Button/Button";
 
 type UnitsTabProps = {
   data: CurriculumUnitsTabData;
 };
 
-const SequenceTab: FC<UnitsTabProps> = ({ data }) => {
-  const { units, threads } = data;
+const UnitsTab: FC<UnitsTabProps> = ({ data }) => {
+  type Unit = CurriculumUnitsTabData["units"][number];
+
+  interface Subject {
+    subject: string;
+    subject_slug: string;
+  }
+
+  interface Domain {
+    title: string;
+    tag_id: number | null;
+  }
+
+  interface Tier {
+    tier: string;
+    tier_slug: string;
+  }
+
+  interface YearSelection {
+    [key: string]: {
+      subject?: Subject | null;
+      domain?: Domain | null;
+      tier?: Tier | null;
+    };
+  }
+
+  const yearData: {
+    [key: string]: {
+      units: Unit[];
+      childSubjects: Subject[];
+      domains: Domain[];
+      tiers: Tier[];
+    };
+  } = {};
+
+  data.units.forEach((unit) => {
+    let data = yearData[unit.year];
+    if (!data) {
+      data = {
+        units: [],
+        childSubjects: [],
+        domains: [],
+        tiers: [],
+      };
+      yearData[unit.year] = data;
+    }
+    data.units.push(unit);
+    const domain = unit.domains[0];
+    if (
+      unit.subject_parent &&
+      unit.subject_parent_slug &&
+      data.childSubjects.every((c) => c.subject_slug !== unit.subject_slug)
+    ) {
+      data.childSubjects.push({
+        subject: unit.subject,
+        subject_slug: unit.subject_slug,
+      });
+    }
+    if (domain && data.domains.every((d) => d.tag_id !== domain.tag_id)) {
+      data.domains.push(domain);
+    }
+    if (
+      unit.tier &&
+      unit.tier_slug &&
+      data.tiers.every((t) => t.tier_slug !== unit.tier_slug)
+    ) {
+      data.tiers.push({
+        tier: unit.tier,
+        tier_slug: unit.tier_slug,
+      });
+    }
+  });
+
+  const initialYearSelection = {} as YearSelection;
+  Object.keys(yearData).forEach((year) => {
+    const data = yearData[year];
+    if (!data) {
+      throw new Error("year data missing");
+    }
+    data.childSubjects.sort((a, b) => {
+      if (a.subject_slug === "combined-science") {
```

Comment:
Is there a way this sort can be achieved on the backend? i.e. not referencing a specific subject slug in this codebase?

Reply:
Good point, will address in the MV and create a different PR for it if that's alright